### PR TITLE
Mon 11764  mon 12394  mon 12695 21.10x backpport

### PR DESCRIPTION
**Pull request by jean-christophe81**

### CODE REVIEW

File: centreon-connector/CMakeLists.txt [hunk 1]
```diff
@@ -1,27 +1,37 @@
...
 
 include(${CMAKE_BINARY_DIR}/conan_paths.cmake)
 find_package(fmt REQUIRED)
 find_package(spdlog REQUIRED)
+find_package(asio REQUIRED)
+find_package(absl REQUIRED)
+find_package(Boost REQUIRED)
+find_package(Libssh2 REQUIRED)
+find_package(OpenSSL REQUIRED)
+find_package(ZLIB REQUIRED)
```
Comment (bouda1) [inline centreon-connector/CMakeLists.txt:25]: Are they needed ?

File: centreon-connector/common/inc/com/centreon/connector/parser.hh [hunk 2]
```diff
@@ -0,0 +1,76 @@
+/*
+** Copyright 2011-2013 Centreon
```
Comment (bouda1) [inline centreon-connector/common/inc/com/centreon/connector/parser.hh:2]: This file looks new :smile: 
So, we can make the copyright to 2022

Comment (bouda1) [reply centreon-connector/common/inc/com/centreon/connector/parser.hh:2]: In fact, it is just moved..., so we just have to complete the date

File: centreon-connector/perl/src/embedded_perl.cc [hunk 3]
```diff
@@ -231,8 +226,8 @@ pid_t embedded_perl::run(std::string const& cmd, int fds[3]) {
     call_pv("Embed::Persistent::run_file", G_DISCARD);
     std::cerr << "error while executing Perl script '" << file
               << "': " << SvPV_nolen(ERRSV) << std::endl;
-    exit(3);
-    abort();
+    //    exit(3);
+    //   abort();
```
Comment (bouda1) [inline centreon-connector/perl/src/embedded_perl.cc:230]: Is it wanted?
* Did you forget to remove comments?
* Did you forget to remove these lines?

File: centreon-connector/perl/src/policy.cc [hunk 4]
```diff
@@ -125,80 +143,39 @@ void policy::on_execute(unsigned long long cmd_id,
...
-  std::lock_guard<std::mutex> lock(processing_mutex);
-
-  // Send check result back to monitoring engine.
-  _reporter.send_result(r);
+void policy::start_end_timer(bool final) {
+  _end_timer.expires_from_now(std::chrono::milliseconds(10));
+  _end_timer.async_wait([me = shared_from_this(),
+                         final](const std::error_code& err) {
+    if (!err) {
+      log::core()->trace("il reste {} checks", checks::check::get_nb_check());
```
Comment (bouda1) [inline centreon-connector/perl/src/policy.cc:158]: In english...

**Review by bouda1 — APPROVED** [review]
Just several comments to check...

File: centreon-connector/CMakeLists.txt (hunk 1 above)
Comment (jean-christophe81) [reply centreon-connector/CMakeLists.txt:25]: yes

File: centreon-connector/perl/src/embedded_perl.cc (hunk 3 above)
Comment (jean-christophe81) [reply centreon-connector/perl/src/embedded_perl.cc]: done

File: centreon-connector/perl/src/policy.cc (hunk 4 above)
Comment (jean-christophe81) [reply centreon-connector/perl/src/policy.cc]: done

File: centreon-connector/common/inc/com/centreon/connector/parser.hh (hunk 2 above)
Comment (jean-christophe81) [reply centreon-connector/common/inc/com/centreon/connector/parser.hh]: done

**Pull request merged**